groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyLanguageServerConfig.java

```diff
@@ -70,13 +81,29 @@ public void setTelemetryEvent(Consumer<Object> telemetryEvent) {
     }
 
     @Override
-    public Consumer<PublishDiagnosticsParams> getPublishDiagnostics() {
-        return publishDiagnostics;
+    public void setPublishDiagnostics(Consumer<PublishDiagnosticsParams> callback) {
+        this.publishDiagnostics = callback;
     }
 
     @Override
-    public void setPublishDiagnostics(Consumer<PublishDiagnosticsParams> callback) {
-        this.publishDiagnostics = callback;
+    public void publishDiagnostics(Set<Diagnostic> diagnostics) {
+        if (diagnostics.isEmpty()) {
+            return;
+        }
+        Map<String, PublishDiagnosticsParamsBuilder> diagnosticsByFile = Maps.newHashMap();
+
+        diagnostics.forEach(diagnostic -> {
+            try {
+                String uri = Uris.absolutePathAsFileUri(Paths.get(diagnostic.getSource()));
```

Comment:
I think we can assume the diagnostic source is always a string in valid uri format